### ISSUE TITLE
z-scores: Simplify distribution query

### ISSUE THREAD

**Issue by austin3dickey** [conbench/conbench#1051]
>Another note from the meeting: for code readability, it would probably make sense to pass along the `contender_benchmark_results` instead of `contender_run_id` and `case_id` and `context_id`. But "filtering the distribution query to a potentially-large python list of case/context pairs" seems hard to do in SQLAlchemy. That's why I didn't do it before.
>
>What if we didn't filter case/context at all in the distribution query, and instead did it in pandas after execution? That could probably speed up the query since we'd remove the complicated join!

_Originally posted by @austin3dickey in https://github.com/conbench/conbench/pull/1049#discussion_r1158664489_

**Comment by jgehrcke** [comment]
>  it would probably make sense to pass along the contender_benchmark_results instead of contender_run_id and case_id and context_id

Yes, cool! Added another high-level thought about this, here: https://github.com/conbench/conbench/pull/1049#discussion_r1159406417.

In my mental model "pass along the contender_benchmark_results instead of " is precisely supporting the following simplification:
- more benchmark result-centric thinking and architecture
- making these group constructs (run, batch) thin or virtual
